lang/resources/org/partiql/type-domains/partiql.ion

```diff
@@ -677,16 +677,45 @@ may then be further optimized by selecting better implementations of each operat
             )
         )
 
-        // Nodes excluded below this line will eventually have a representation in the logical algebra, but not
-        // initially.
+        (include
+            // Indicates kind of DML operation.
+            (sum dml_operation
+                (dml_insert)
+                (dml_delete)
+                // TODO:
+                // (dml_update)
+            )
+        )
 
+        // Redefine statement.dml to be a simpler subset of the full DML functionality expressed with PartiQL's DML
+        // syntax.  Full functionality is out of scope for now.  This is factored minimally to support
+        // `INSERT INTO` and `DELETE FROM ... [WHERE <predicate>]` but this may need refactoring when
+        // `FROM ... UPDATE` and `UPDATE` is supported later.
         (with statement
-            (exclude
-                dml
-                ddl
+            (exclude dml)
+            (include
+                // A DML operation, such as `INSERT`, `UPDATE` or `DELETE`
+                (dml
+                    // The target is an expression that is indicates the table whose data is to be manipulated.
+                    // In practice this can be an identifier or a simplified path expression consisting of only
+                    // literal path steps (and with no wildcard or unpivot operators).  This restriction must
+                    // be enforced by the PartiQL implementation.
+                    // TODO: consider ways to reduce the domain of acceptable values here.  Of course, we it would
+                    // be possible to create a `sum` type like `expr` but containing only the valid DML targets,
+                    // but that would might necessitate duplicating some logic related to variable resolution which is
+                    // already implemented for standard expressions.  This is out of scope for current efforts.
+                    target::expr
```

Comment:
I understand that we have a comment and a `TODO` for this, but could we go w/ the restrictive approach and use an `identifier` for now? With the current spec, we're target `<table name> ::= <identifier>` to limit the scope.

Reply:
How about we create an issue and I address it next thing?  This PR is already pretty large and I would need to think about how to avoid the mentioned copy-pasta.

Reply:
Sounds good, thanks.

Reply:
https://github.com/partiql/partiql-lang-kotlin/issues/677